service/resource/certificate/desired.go

```diff
@@ -2,8 +2,67 @@ package certificate
 
 import (
 	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/prometheus-config-controller/service/key"
+	"github.com/giantswarm/prometheus-config-controller/service/prometheus"
+)
+
+const (
+	CaKey  = "ca"  // CaKey is the key in the Secret that holds the CA.
+	CrtKey = "crt" // CrtKey is the key in the Secret that holds the certificate.
+	KeyKey = "key" // KeyKey is the key in the Secret that holds the key.
 )
 
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	r.logger.Log("debug", "fetching all services")
+	services, err := r.k8sClient.CoreV1().Services("").List(metav1.ListOptions{})
+	if err != nil {
+		return nil, microerror.Maskf(err, "an error occurred listing all services")
+	}
+
+	r.logger.Log("debug", "filtering services")
+	validServices := prometheus.FilterInvalidServices(services.Items)
+
+	r.logger.Log("debug", "fetching certificates")
+	certificateFiles := []certificateFile{}
+	for _, service := range validServices {
+		namespace, name := prometheus.GetCertificateName(service)
+
+		certificate, err := r.k8sClient.CoreV1().Secrets(namespace).Get(name, metav1.GetOptions{})
```

Comment:
All this secret watching/fetching is already provided by the cert watcher. See https://github.com/giantswarm/certificatetpr/blob/02983fbbe2165d6acd9e6cb081ace6769311d67a/service.go#L69. IMO we should use this instead of reinventing it again.